Code/Editor/MainWindow.cpp

```diff
@@ -845,78 +852,190 @@ void MainWindow::InitActions()
+        .SetToolTip(tr("Location 7 (Shift+F7)"))
+        .Connect(&QAction::triggered, [goToViewBookmarkFn](){ goToViewBookmarkFn(6);});
     am->AddAction(ID_GOTO_LOC8, tr("Location 8"))
         .SetShortcut(tr("Shift+F8"))
-        .SetToolTip(tr("Location 8 (Shift+F8)"));
+        .SetToolTip(tr("Location 8 (Shift+F8)"))
+        .Connect(&QAction::triggered, [goToViewBookmarkFn](){ goToViewBookmarkFn(7);});
     am->AddAction(ID_GOTO_LOC9, tr("Location 9"))
         .SetShortcut(tr("Shift+F9"))
-        .SetToolTip(tr("Location 9 (Shift+F9)"));
+        .SetToolTip(tr("Location 9 (Shift+F9)"))
+        .Connect(&QAction::triggered, [goToViewBookmarkFn](){ goToViewBookmarkFn(8);});
     am->AddAction(ID_GOTO_LOC10, tr("Location 10"))
         .SetShortcut(tr("Shift+F10"))
-        .SetToolTip(tr("Location 10 (Shift+F10)"));
+        .SetToolTip(tr("Location 10 (Shift+F10)"))
+        .Connect(&QAction::triggered, [goToViewBookmarkFn](){ goToViewBookmarkFn(9);});
     am->AddAction(ID_GOTO_LOC11, tr("Location 11"))
         .SetShortcut(tr("Shift+F11"))
-        .SetToolTip(tr("Location 11 (Shift+F11)"));
+        .SetToolTip(tr("Location 11 (Shift+F11)"))
+        .Connect(&QAction::triggered, [goToViewBookmarkFn](){ goToViewBookmarkFn(10);});
     am->AddAction(ID_GOTO_LOC12, tr("Location 12"))
         .SetShortcut(tr("Shift+F12"))
-        .SetToolTip(tr("Location 12 (Shift+F12)"));
+        .SetToolTip(tr("Location 12 (Shift+F12)"))
+        .Connect(&QAction::triggered, [goToViewBookmarkFn](){ goToViewBookmarkFn(11);});
+
+    const auto tagViewBookmarkFn = [](int index)
+    {
+        if (AzToolsFramework::ViewBookmarkLoaderInterface* bookmarkLoader = AZ::Interface<ViewBookmarkLoaderInterface>::Get())
+        {
+            bool found = false;
+            AzFramework::CameraState cameraState;
+            Camera::EditorCameraRequestBus::BroadcastResult(
+                found, &Camera::EditorCameraRequestBus::Events::GetActiveCameraState, cameraState);
+            if (found)
+            {
+                ViewBookmark bookmark;
+                bookmark.m_position = cameraState.m_position;
+                bookmark.m_rotation =
+                    AzFramework::EulerAngles(AZ::Matrix3x3::CreateFromColumns(cameraState.m_side, cameraState.m_forward, cameraState.m_up));
+
+                bookmarkLoader->ModifyBookmarkAtIndex(bookmark, index, ViewBookmarkLoaderInterface::StorageMode::Local);
+                QString tagConsoleText = tr("View Bookmark %1 set to the position: x=%2, y=%3, z=%4")
+                                             .arg(index + 1)
+                                             .arg(bookmark.m_position.GetX(), 0, 'f', 2)
+                                             .arg(bookmark.m_position.GetY(), 0, 'f', 2)
+                                             .arg(bookmark.m_position.GetZ(), 0, 'f', 2);
+                GetIEditor()->WriteToConsole(tagConsoleText.toUtf8().data());
```

Comment:
suggestion: To make this code easier to port in future it might be worth switching the `GetIEditor()->WriteToConsole` calls to `AZ_Printf`, should have the same effect and not have to reference `IEditor`